support/SubmitScripts/OceanClang.sh

```diff
@@ -76,13 +76,28 @@ for node in $(echo $SLURM_NODELIST | scontrol show hostnames); do
   echo "host ${node}" >> nodelist.$SLURM_JOBID
 done
 
-# The 19 is there because Charm++ uses one thread per node for communication
-# Here, -np should take the number of nodes (must be the same as --nodes
+# The (SLURM_NTASKS_PER_NODE - 1) is there because Charm++ uses one thread per
+# node for communication
+# Here, ++np should take the number of nodes (must be the same as --nodes
 # in the #SBATCH options above).
 WORKER_THREADS_PER_NODE=$((SLURM_NTASKS_PER_NODE - 1))
 WORKER_THREADS=$((SLURM_NPROCS - SLURM_NNODES))
 SPECTRE_COMMAND="${SPECTRE_EXECUTABLE} ++np ${SLURM_NNODES} \
 ++p ${WORKER_THREADS} ++ppn ${WORKER_THREADS_PER_NODE} \
 ++nodelist nodelist.${SLURM_JOBID}"
 
-charmrun ${SPECTRE_COMMAND} --input-file ${SPECTRE_INPUT_FILE}
+
+# When invoking through `charmrun`, charm will initiate remote sessions which
+# will wipe out environment settings unless it is forced to re-initialize the
+# spectre environment between the start of the remote session and starting the
+# spectre executable
+echo "#!/bin/sh
+source /home/moxon/spectre/support/Environments/ocean_clang.sh
+spectre_load_modules
+\$@
+" > runscript
```

Comment:
Is there a way to write this into a tmp directory or with a random suffix? I'm worried that this will subtly overwrite someone's script/files and cause all kinds of havoc

Reply:
hmm.. fair point. I see how 'runscript' has some possibility of colliding with something :P.
I've appended the slurm job id like we do with the nodelist.

Reply:
That's reasonable. Should it be deleted at the end of the job? I'm undecided on that... It's nice to have a record of exactly what went down :shrug:

Reply:
Yeah, I considered removing it, but I think it's more helpful to save the script (easier when trying to figure out what's going wrong if something changes down the line).

Reply:
:+1: that sounds good to me :)